airflow/providers/amazon/aws/sensors/s3_upload_session_complete.py

```diff
@@ -0,0 +1,172 @@
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+from datetime import datetime
+from typing import Optional, Set, Union
+
+from cached_property import cached_property
+
+from airflow.exceptions import AirflowException
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+from airflow.sensors.base_sensor_operator import BaseSensorOperator, poke_mode_only
+from airflow.utils.decorators import apply_defaults
+
+
+@poke_mode_only
+class S3UploadSessionCompleteSensor(BaseSensorOperator):
+    """
+    Checks for changes in the number of objects at prefix in AWS S3
+    bucket and returns True if the inactivity period has passed with no
+    increase in the number of objects. Note, this sensor will not behave correctly
+    in reschedule mode, as the state of the listed objects in the S3 bucket will
+    be lost between rescheduled invocations.
+
+    :param bucket_name: Name of the S3 bucket
+    :type bucket_name: str
+    :param prefix: The prefix being waited on. Relative path from bucket root level.
+    :type prefix: str
+    :param aws_conn_id: a reference to the s3 connection
+    :type aws_conn_id: str
+    :param verify: Whether or not to verify SSL certificates for S3 connection.
+        By default SSL certificates are verified.
+        You can provide the following values:
+
+        - ``False``: do not validate SSL certificates. SSL will still be used
+                 (unless use_ssl is False), but SSL certificates will not be
+                 verified.
+        - ``path/to/cert/bundle.pem``: A filename of the CA cert bundle to uses.
+                 You can specify this argument if you want to use a different
+                 CA cert bundle than the one used by botocore.
+    :type verify: bool or str
+    :param inactivity_period: The total seconds of inactivity to designate
+        an upload session is over. Note, this mechanism is not real time and
+        this operator may not return until a poke_interval after this period
+        has passed with no additional objects sensed.
+    :type inactivity_period: float
+    :param min_objects: The minimum number of objects needed for upload session
+        to be considered valid.
+    :type min_objects: int
+    :param previous_objects: The set of object ids found during the last poke.
+    :type previous_objects: set[str]
```

Comment:
```suggestion
    :type previous_objects: Optional[Set[str]]
```
You can use `Optional` in the docs, too.